### ISSUE TITLE
Change show instances to output ISO format when possible

### ISSUE THREAD

**Issue by mlopes** [mlopes/wen#3]

**Comment by asarturas** [comment]
you mean ISO-8601 with milliseconds? like `2019-01-01T01:01:01.001Z`

**Comment by mlopes** [comment]
Yes, but not just for ZoneDateTime, for all applicable formats.
Here are the ISO formats for dates and times:

https://en.m.wikipedia.org/wiki/ISO_8601

Also, the standard doesn’t natively support dates before 1582, so I propose, keeping the same format up to 1 AD and then just add BC as a suffix to the year.

They do propose an extension to the ISO format to be agreed by the date provider and user, which is a bit messy because it adds characters to the year, and a sign, and shift all BC dates by one by making year 0000 be 1 BC. Since in a library you can’t really agree with the consumers I suggest ignoring that and just using the standard and the BC suffix.

**Comment by asarturas** [comment]
Would you prefer to keep existing verbose version or completely replace show with iso?

**Comment by mlopes** [comment]
I think we could drop the current version, it's an arbitrary format, although close enough to ISO, it probably just needs the space between date and time to become `T` and if it is UTC put a `Z` in the end instead of `+00:00`. Also it needs the `AD` to be removed from the string, when dates are 1 AD or after.

TL;DR arbitrary format that has no use if we can have ISO, it should go

**Comment by mlopes** [comment]
2019-05-04
2019-05-04T15:06:04+01:00
2019-05-04T15:06:04Z
13:47:30.042
0004BC-05-04
2004BC-05-04T15:06:04Z

**Comment by mlopes** [comment]
Scope change after discussion with @asarturas . Instead of replacing the default instance, this will have its own instance, possibly `wen.instances.iso` instead of replacing the human readable ones.
